// angular-froala-wysiwyg/esm2015/fe-root.module.js
import { NgModule } from '@angular/core';
import { FroalaEditorModule } from './editor/editor.module';
import { FroalaViewModule } from './view/view.module';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './editor/editor.module';
import * as ɵngcc2 from './view/view.module';
export class FERootModule {
}
FERootModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: FERootModule });
FERootModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function FERootModule_Factory(t) { return new (t || FERootModule)(); }, imports: [[
            FroalaEditorModule.forRoot(),
            FroalaViewModule.forRoot()
        ], FroalaEditorModule,
        FroalaViewModule] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(FERootModule, { imports: function () { return [ɵngcc1.FroalaEditorModule, ɵngcc2.FroalaViewModule]; }, exports: function () { return [FroalaEditorModule,
        FroalaViewModule]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(FERootModule, [{
        type: NgModule,
        args: [{
                imports: [
                    FroalaEditorModule.forRoot(),
                    FroalaViewModule.forRoot()
                ],
                exports: [
                    FroalaEditorModule,
                    FroalaViewModule
                ]
            }]
    }], null, null); })();
