// ngx-font-awesome/esm5/lib/ngx-font-awesome.module.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { NgModule } from '@angular/core';
import { NgxFontAwesomeComponent } from './ngx-font-awesome.component';
import { CommonModule } from '@angular/common';
import * as ɵngcc0 from '@angular/core';
var NgxFontAwesomeModule = /** @class */ (function () {
    function NgxFontAwesomeModule() {
    }
NgxFontAwesomeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgxFontAwesomeModule });
NgxFontAwesomeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgxFontAwesomeModule_Factory(t) { return new (t || NgxFontAwesomeModule)(); }, imports: [[
            CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgxFontAwesomeModule, { declarations: function () { return [NgxFontAwesomeComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [NgxFontAwesomeComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgxFontAwesomeModule, [{
        type: NgModule,
        args: [{
                declarations: [NgxFontAwesomeComponent],
                imports: [
                    CommonModule
                ],
                exports: [NgxFontAwesomeComponent]
            }]
    }], function () { return []; }, null); })();
    return NgxFontAwesomeModule;
}());
export { NgxFontAwesomeModule };
