// angular-froala-wysiwyg/esm2015/editor/editor.module.js
import { NgModule } from '@angular/core';
import { FroalaEditorDirective } from './editor.directive';
import * as ɵngcc0 from '@angular/core';
export class FroalaEditorModule {
    static forRoot() {
        return { ngModule: FroalaEditorModule, providers: [] };
    }
}
FroalaEditorModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: FroalaEditorModule });
FroalaEditorModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function FroalaEditorModule_Factory(t) { return new (t || FroalaEditorModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(FroalaEditorModule, { declarations: function () { return [FroalaEditorDirective]; }, exports: function () { return [FroalaEditorDirective]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(FroalaEditorModule, [{
        type: NgModule,
        args: [{
                declarations: [FroalaEditorDirective],
                exports: [FroalaEditorDirective]
            }]
    }], null, null); })();
